src/modules/dashboard/components/ActionsPage/DetailsWidget/DetailsWidgetSafeTransaction/components.tsx

```diff
@@ -0,0 +1,396 @@
+import { Placement } from '@popperjs/core';
+import React from 'react';
+import { defineMessage, FormattedMessage, MessageDescriptor } from 'react-intl';
+import classnames from 'classnames';
+import { nanoid } from 'nanoid';
+
+import DetailsWidgetUser from '~core/DetailsWidgetUser';
+import Icon from '~core/Icon';
+import InvisibleCopyableAddress from '~core/InvisibleCopyableAddress';
+import MaskedAddress from '~core/MaskedAddress';
+import Avatar from '~core/Avatar';
+import Numeral from '~core/Numeral';
+import AddressDetailsView from '~dashboard/Dialogs/ControlSafeDialog/TransactionPreview/AddressDetailsView';
+import { TransactionTypes } from '~dashboard/Dialogs/ControlSafeDialog/constants';
+import { defaultTransaction } from '~dashboard/Dialogs/ControlSafeDialog/ControlSafeDialog';
+import {
+  SafeTransaction,
+  ColonySafe,
+  SafeBalanceToken,
+  Erc20Token,
+} from '~data/generated';
+import { AnyUser, Colony } from '~data/index';
+import { intl } from '~utils/intl';
+import { getArrayFromString } from '~utils/safes';
+import {
+  extractParameterName,
+  extractParameterType,
+} from '~utils/safes/getContractUsefulMethods';
+import { omit } from '~utils/lodash';
+
+import { toRecipientMSG, valueMSG } from '../DetailsWidget';
+
+import widgetStyles from '../DetailsWidget.css';
+import styles from './DetailsWidgetSafeTransaction.css';
+
+const MSG = defineMessage({
+  contract: {
+    id: `dashboard.ActionsPage.DetailsWidget.DetailsWidgetSafeTransaction.components.contract`,
```

Comment:
`.components.` is not really a React component, just a grouping folder so therefore it should not be included in the ids of these messages